app/models/maintenance_tasks/task.rb

```diff
@@ -175,7 +175,17 @@ def load_constants
         namespace = MaintenanceTasks.tasks_module.safe_constantize
         return unless namespace
 
-        namespace.constants.map { |constant| namespace.const_get(constant) }
+        namespace.constants
+          .map { |constant| namespace.const_get(constant) }
+          .filter { |klass| can_initialize?(klass) }
```

Comment:
Also, checking for `#abstract!` is not 100%, because non-abstract classes might use `T::Helpers`:

```ruby
irb(main):001:1* class SomethingElse
irb(main):002:1*   extend T::Helpers
irb(main):003:0> end
=> SomethingElse
irb(main):004:0> SomethingElse.respond_to?(:abstract!)
=> true
```

Reply:
Using sorbet-provided methods is the least preferable option as we don't want the gem to have `sorbet` as a dependency. The gem should have an abstraction to exclude classes that are not supposed to be a runnable maintenance task. Currently the only option is to use a different namespace for the abstract task so it's not being picked up by the gem. But I don't think that's the best recommendation and the gem itself can benefit from an improvement as currently for example `Maintenance::SharedModuleBetweenTasks` will also be picked up as a maintenance task while not being a module and focusing on sorbet-specific problem won't solve it for other use-cases.

Maybe modules can be filtered out by the gem itself and the app can define `Maintenance::AbstractTaskInterface` as a sorbet interface?  

But if we still want to use inheritance, perhaps the gem should have a configurable filter `proc` and each app can be defining it's own set of filters to exclude certain constants from being treated as tasks. It's going to be up to the app whether to use sorbet provided helpers or just hardcode constant name in the filter. 
And the gem itself should stay abstracted from all these details and use-cases